### DISCUSSION TITLE
Sync-PnPSharePointUserProfilesFromAzureActiveDirectory - sync extension attributes

### DISCUSSION THREAD

**Discussion by r998877** [pnp/powershell#3231] [Q&A]
Hello all,

does someone know, how to synchronize extension attributes from Azure AD to SharePoint User Profile services with Sync-PnPSharePointUserProfilesFromAzureActiveDirectory?

https://pnp.github.io/powershell/cmdlets/Sync-PnPSharePointUserProfilesFromAzureActiveDirectory.html#example-2

In Example 2 the following mapping is used:

$users = Get-PnPAzureADUser -Filter "jobTitle eq 'IT Administrator'"
Sync-PnPSharePointUserProfilesFromAzureActiveDirectory -UserProfilePropertyMapping @{"CostCenter"="extension_**b0b5aaa58a0a4287acd826c5b8330e48**_CostCenter"} -Users $users

The part marked in bold seems to be some sort of tenant specific GUID. What kind of GUID is this?

Regards,
r

**Comment by AndersRask** [comment]
First of all for those extension attributes to show up, either ADConnect or Exchange need to be set up to sync to AAD, because this will extend the schema with those properties.

The guid that comes after extension_  are unique to your installation, so you will need to look them up, either using PowerShell or in AAD in the portal.

In the portal go to AAD > Users > select a user > select the Properties tab > look for Extension attributes in the last column.

With PowerShell use graph, AzureAD or Az module to get a user.
The extension attributes are located as a hashtable under ExtensionProperty

hth
Anders
